crates/matrix-sdk-indexeddb/src/media_store/types.rs

```diff
@@ -37,28 +37,50 @@ impl Lease {
     }
 }
 
-/// A representation of media data which can be stored in IndexedDB.
-#[derive(Debug, Serialize, Deserialize)]
+/// A representation of media which ignores storage schemas. This is type is not
+/// stored in IndexedDB, and is mostly useful for passing media around which may
+/// eventually be transformed into types which are storable in IndexedDB.
+#[derive(Debug)]
 pub struct Media {
-    /// The metadata associated with [`Media::content`]
-    pub metadata: MediaMetadata,
+    /// The parameters specifying the type and source of the media contained in
+    /// [`Media::content`]
+    pub request_parameters: MediaRequestParameters,
+    /// The last time the media was accessed in IndexedDB
+    pub last_access: UnixTime,
+    /// Whether to ignore the [`MediaRetentionPolicy`][1] stored in IndexedDB
+    ///
+    /// [1]: matrix_sdk_base::media::store::MediaRetentionPolicy
+    pub ignore_policy: IgnoreMediaRetentionPolicy,
     /// The content of the media
     pub content: Vec<u8>,
 }
 
 /// A representation of media metadata which can be stored in IndexedDB.
 #[derive(Debug, Serialize, Deserialize)]
 pub struct MediaMetadata {
-    /// The parameters specifying the type and source of the media contained in
-    /// [`Media::content`]
+    /// The parameters specifying the type and source of the associated
+    /// [`MediaContent`]
     pub request_parameters: MediaRequestParameters,
-    /// The last time the media was accessed in IndexedDB
+    /// The last time the associated [`MediaContent`] was accessed in IndexedDB
     pub last_access: UnixTime,
     /// Whether to ignore the [`MediaRetentionPolicy`][1] stored in IndexedDB
     ///
     /// [1]: matrix_sdk_base::media::store::MediaRetentionPolicy
     #[serde(with = "crate::media_store::serializer::foreign::ignore_media_retention_policy")]
     pub ignore_policy: IgnoreMediaRetentionPolicy,
+    /// The identifier of the associated [`MediaContent`]
+    pub content_id: u64,
+    /// The size in bytes of the associated [`MediaContent`]
+    pub content_size: usize,
+}
+
+/// A representation of media content which can be stored in IndexedDB.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct MediaContent {
+    /// The identifier associated with the given [`MediaContent::data`].
+    pub id: u64,
```

Comment:
As far as I gather `MediaMetadata::content_id` and `MediaContent::id` are the unique key which is used to reference each other from one type to the other.

Can we name the fields the same and perhaps add a type alias or even wrapper type for the `u64` to make this a bit more obvious?

Reply:
Just making sure I understand correctly: you are asking that `MediaMetadata::content_id` and `MediaContent::id` have the same name and use `IndexedMediaContentIdKey` as their type. Is that correct?

This seems fine to me and, in fact, I even had it that way at one point! How would you feel about them both being named `content_id`? While `MediaContent::content_id` feels redundant, I think it is preferable to `MediaMetadata::id`, which seems far too general to indicate that it is actually a type of foreign key.

Or if you have another suggestion that is better than either of those, happy to hear that as well.